backend/core/src/main/java/org/sonarsource/sonarlint/core/rules/RulesService.java

```diff
@@ -303,10 +316,43 @@ public void onServerEventReceived(SonarServerEventReceivedEvent eventReceived) {
     var connectionId = eventReceived.getConnectionId();
     var serverEvent = eventReceived.getEvent();
     if (serverEvent instanceof RuleSetChangedEvent) {
-      updateStorage(connectionId, (RuleSetChangedEvent) serverEvent);
+      var ruleSetChangedEvent = (RuleSetChangedEvent) serverEvent;
+      updateStorage(connectionId, ruleSetChangedEvent);
+      processEvent(ruleSetChangedEvent);
     }
   }
 
+  private void processEvent(RuleSetChangedEvent event) {
+    var deactivatedRules = event.getDeactivatedRules();
```

Comment:
We could also leave a comment, or open a ticket to trigger an analysis when rules were enabled